modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/XmlExtensionParserUtils.java

```diff
@@ -0,0 +1,491 @@
+import org.mule.config.spring.MuleHierarchicalBeanDefinitionParserDelegate;
+import org.mule.config.spring.parsers.generic.AutoIdUtils;
+import org.mule.extension.introspection.DataQualifier;
+import org.mule.extension.introspection.DataQualifierVisitor;
+import org.mule.extension.introspection.DataType;
+import org.mule.extension.introspection.Parameter;
+import org.mule.module.extension.internal.capability.xml.schema.model.SchemaConstants;
+import org.mule.module.extension.internal.introspection.BaseDataQualifierVisitor;
+import org.mule.module.extension.internal.introspection.SimpleTypeDataQualifierVisitor;
+import org.mule.module.extension.internal.runtime.DefaultObjectBuilder;
+import org.mule.module.extension.internal.runtime.ObjectBuilder;
+import org.mule.module.extension.internal.runtime.resolver.CachingValueResolverWrapper;
+import org.mule.module.extension.internal.runtime.resolver.CollectionValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.EvaluateAndTransformValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.NestedProcessorValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.ObjectBuilderValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.RegistryLookupValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.ResolverSet;
+import org.mule.module.extension.internal.runtime.resolver.StaticValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.ValueResolver;
+import org.mule.module.extension.internal.util.IntrospectionUtils;
+import org.mule.util.TemplateParser;
+import org.mule.util.ValueHolder;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.lang.reflect.Field;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.ManagedList;
+import org.springframework.beans.factory.support.ManagedMap;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.support.DefaultConversionService;
+import org.springframework.util.xml.DomUtils;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+
+/**
+ * Utility methods for XML parsers capable of handling objects described by the extensions introspection API
+ *
+ * @since 3.7.0
+ */
+final class XmlExtensionParserUtils
+{
+
+    private static final String DATE_FORMAT = "yyyy-MM-dd'T'hh:mm:ss";
+    private static final String CALENDAR_FORMAT = "yyyy-MM-dd'T'hh:mm:ssX";
+
+    private static final TemplateParser parser = TemplateParser.createMuleStyleParser();
```

Comment:
This class is not thread safe. An utility class should not use static fields that are not thread safe. Maybe this class must be instantiated